src/library/scala/collection/Iterator.scala

```diff
@@ -1189,6 +1223,80 @@ object Iterator extends IterableFactory[Iterator] {
       } else Iterator.empty.next()
     }
   }
+
+  private final class TakeRightIterator[A](private[this] var underlying: Iterator[A], buflen: Int) extends AbstractIterator[A] {
+    private[this] var len: Int = -1
+    private[this] var pos: Int = 0
+    private[this] var buf: Array[AnyRef] = _
+    def init(): Unit = if(buf eq null) {
+      buf = new Array[AnyRef](buflen)
```

Comment:
Does @Ichoran's https://github.com/scala/collection-strawman/pull/96#discussion_r120459949 apply here as well?

Reply:
Yes, it should apply!  Preallocating when you don't actually know that anywhere near that much space is needed isn't a good general-purpose strategy, even if it is more efficient in the case where the bound is chosen wisely.